pkg/controller/bypass/bypass_controller.go

```diff
@@ -0,0 +1,277 @@
+package bypass
+
+import (
+	"bytes"
+	"context"
+	"embed"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	netns "github.com/containernetworking/plugins/pkg/ns"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+
+	nd "istio.io/istio/cni/pkg/nodeagent"
+	"istio.io/istio/pkg/util/sets"
+	"kmesh.net/kmesh/pkg/logger"
+	"kmesh.net/kmesh/pkg/utils"
+)
+
+var (
+	log = logger.NewLoggerField("bypass")
+	FS  embed.FS
+)
+
+func getCurrentNodeName(client kubernetes.Interface, podName, namespace string) (string, error) {
+	pod, err := client.CoreV1().Pods(namespace).Get(context.Background(), podName, metav1.GetOptions{})
```

Comment:
this is accessing the pod directly from k8s. overload kube-apiserver

There is downward API, which can directly pass node Name as an pod env

Reply:
- name: NODE_NAME
              valueFrom:
                fieldRef:
                  fieldPath: spec.nodeName